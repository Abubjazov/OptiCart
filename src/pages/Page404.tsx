import { Link } from 'react-router-dom'
import { ErrorMessage } from '../components'

export const Page404 = (): JSX.Element => (
	<main>
		<ErrorMessage />
		<p style={{ textAlign: 'center', fontWeight: 300, fontSize: 24 }}>
			Page doesn't exist
		</p>
		<Link
			to='/'
			style={{
				display: 'block',
				textAlign: 'center',
				fontWeight: 300,
				fontSize: 24,
				marginTop: 13,
				color: '#429188',
			}}
		>
			Back to main page
		</Link>
	</main>
)
