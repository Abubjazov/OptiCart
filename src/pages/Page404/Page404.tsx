import { ErrorGif } from '../../components'

export const Page404 = (): JSX.Element => (
	<main
		style={{
			paddingTop: 100,
		}}
	>
		<ErrorGif />
		<p
			tabIndex={0}
			role='alert'
			style={{ textAlign: 'center', fontWeight: 300, fontSize: 24 }}
		>
			Page doesn't exist
		</p>
		<a
			href='/optiCart/'
			role='alert'
			style={{
				display: 'block',
				textAlign: 'center',
				fontWeight: 300,
				fontSize: 24,
				marginTop: 13,
				color: '#3b8079',
			}}
		>
			Back to main page
		</a>
	</main>
)
