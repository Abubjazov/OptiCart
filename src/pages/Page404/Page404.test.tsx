import { render } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'

import { Page404 } from './Page404'

describe('Component: MainPage', () => {
	test('should render MainPage', () => {
		const { asFragment } = render(
			<BrowserRouter>
				<Page404 />
			</BrowserRouter>
		)
		expect(asFragment()).toMatchSnapshot()
	})
})
