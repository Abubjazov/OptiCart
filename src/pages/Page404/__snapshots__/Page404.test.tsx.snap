// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: MainPage should render MainPage 1`] = `
<DocumentFragment>
  <main>
    <img
      alt="An error has occurred"
      src="error.gif"
      style="display: block; width: 250px; height: 250px; object-fit: contain; margin: 0px auto;"
    />
    <p
      style="text-align: center; font-weight: 300; font-size: 24px;"
    >
      Page doesn't exist
    </p>
    <a
      href="/"
      style="display: block; text-align: center; font-weight: 300; font-size: 24px; margin-top: 13px; color: rgb(66, 145, 136);"
    >
      Back to main page
    </a>
  </main>
</DocumentFragment>
`;
